src/tsconfig.json

```diff
@@ -12,5 +12,6 @@
     "compiler/index.ts",
     "value/index.ts",
```

Comment:
Lets include `guards/index.ts` here. This enables TypeScript to emit the code for the guards without importing through `src/index.ts`.